presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestMergeIdenticalWindows.java

```diff
@@ -0,0 +1,150 @@
+                "sum(discount) over (partition by suppkey order by orderkey rows between unbounded preceding and current row) sum_discount_A " +
+                "from lineitem";
+
+        assertUnitPlan(sql,
+                anyTree(
+                        window(ImmutableList.of(functionCall("sum", "quantity"), functionCall("sum", "discount")),
+                                window(ImmutableList.of(functionCall("sum", "quantity")),
+                                        any()))));
+    }
+
+    @Test
+    public void testIdenticalWindowSpecificationsABcpA()
+    {
+        @Language("SQL") String sql = "select " +
+                "sum(quantity) over (partition by suppkey order by orderkey rows between unbounded preceding and current row) sum_quantity_A, " +
+                "lag(quantity, 1, 0.0) over (partition by orderkey order by shipdate rows between UNBOUNDED preceding and CURRENT ROW) sum_quantity_B, " +
+                "sum(discount) over (partition by suppkey order by orderkey rows between unbounded preceding and current row) sum_discount_A " +
+                "from lineitem";
+
+        assertUnitPlan(sql,
+                anyTree(
+                        window(ImmutableList.of(functionCall("sum", "discount")),
+                                window(ImmutableList.of(functionCall("lag", "quantity", "*", "*")),
+                                        project(
+                                                window(ImmutableList.of(functionCall("sum", "quantity")),
+                                                        any()))))));
+    }
+
+    @Test
+    public void testIdenticalWindowSpecificationsAAcpA()
+    {
+        @Language("SQL") String sql = "select " +
+                "sum(quantity) over (partition by suppkey order by orderkey rows between unbounded preceding and current row) sum_quantity_A, " +
+                "lag(quantity, 1, 0.0) over (partition by suppkey order by orderkey rows between UNBOUNDED preceding and CURRENT ROW) sum_quantity_B, " +
+                "sum(discount) over (partition by suppkey order by orderkey rows between unbounded preceding and current row) sum_discount_A " +
+                "from lineitem";
+
+        assertUnitPlan(sql,
+                anyTree(
+                        window(ImmutableList.of(functionCall("sum", "discount"), functionCall("lag", "quantity", "*", "*")),
+                                project(
+                                        window(ImmutableList.of(functionCall("sum", "quantity")),
+                                                any())))));
+    }
+
+    private void assertUnitPlan(@Language("SQL") String sql, PlanMatchPattern pattern)
+    {
+        Plan actualPlan = unitPlan(sql);
+        queryRunner.inTransaction(transactionSession -> {
+            PlanAssert.assertPlan(transactionSession, queryRunner.getMetadata(), actualPlan, pattern);
+            return null;
+        });
+    }
+
+    private Plan unitPlan(@Language("SQL") String sql)
```

Comment:
Small thing to add, we chose to use this approach instead of "true" unit test because it's much more readable with SQL queries instead of building optimizer Nodes manually.